### PR TITLE
Iv/gerrit validate connection fix2

### PR DESCRIPTION
**Pull request by varsanojidan**
Adding a working implementation of the ValidateConnection method

Waiting on https://github.com/sourcegraph/sourcegraph/pull/35749 to be merged first before this.

Closes https://github.com/sourcegraph/sourcegraph/issues/35747

## Test plan

Created unit tests for coverage

### CODE REVIEW

**Comment by mollylogue** [comment]
> I believe you can rebase on main (`git rebase -i main`) and then drop any commits of mine that show up during the interactive rebase and you'll just be left with your commits. It'll likely make this easier to read and determine which changes you're making here.

Or create a new branch off `main` and cherry-pick your commits onto that 🤷‍♀️

**Comment by varsanojidan** [comment]
@mollylogue I merged main into this branch so that the changes are made are separate from the rest of the Gerrit work so it wont show up

**Review by mollylogue — APPROVED** [review]
👍 Looks good!

File: enterprise/internal/authz/gerrit/gerrit.go [hunk 1]
```diff
@@ -134,6 +139,23 @@ func (p Provider) URN() string {
...
-	return nil
+
+	adminGroup, err := p.client.GetGroup(ctx, adminGroupName)
+	if err != nil {
+		return []string{
+			fmt.Sprintf("Unable to get %s group: %v", adminGroupName, err),
+		}
+	}
+
+	if adminGroup.ID == "" || adminGroup.Name != adminGroupName || adminGroup.CreatedOn == "" {
```
[the remark this answers is not in the document]
Comment (varsanojidan) [reply enterprise/internal/authz/gerrit/gerrit.go:154]: Yeah they're always present, so there wouldn't be an issue checking for them

**Pull request merged**